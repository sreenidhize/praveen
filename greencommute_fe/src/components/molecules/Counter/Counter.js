/* eslint-disable no-unused-vars */
import { makeStyles, Typography } from "@material-ui/core";
import React from "react";
import PropTypes from "prop-types";

const useStyles = makeStyles((theme) => ({
  textClass: {
    textAlign: "center",
  },
}));
function Counter(props) {
  const { number, city, showCity, ...rest } = props;
  const classes = useStyles();

  return (
    <>
      <Typography variant="h1" className={classes.textClass} {...rest}>
        {number}
      </Typography>
      {showCity && (
        <Typography variant="h3" {...rest}>
          {city}
        </Typography>
      )}
    </>
  );
}

Counter.propTypes = {
  number: PropTypes.string.isRequired,
  city: PropTypes.string,
  showCity: PropTypes.bool,
};

Counter.defaultProps = {
  city: "",
  showCity: true,
};

export default Counter;
