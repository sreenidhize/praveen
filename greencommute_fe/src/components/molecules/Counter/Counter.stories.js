import React from "react";
import Counter from "./Counter";

const CounterStory = {
  title: "Molecules/Counter",
  component: Counter,
};

const Template = (args) => <Counter {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  number: "953",
  city: "Hyderabad",
  showCity: true,
};

export default CounterStory;
