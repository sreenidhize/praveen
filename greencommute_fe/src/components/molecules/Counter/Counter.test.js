import React from "react";
import Counter from "./Counter";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

describe("Counter renders correctly", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(<Counter number="153" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("Should render Typography", () => {
    const { container } = render(
      <Counter number="953" city="Hyderabad" data-testid="AQI" />
    );
    screen.getAllByTestId("AQI");
    expect(container.querySelector("h1")).toBeInTheDocument();
  });
});
