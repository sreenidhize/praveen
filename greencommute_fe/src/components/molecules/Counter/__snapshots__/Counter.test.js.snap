// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <h1
    class="MuiTypography-root makeStyles-textClass-1 MuiTypography-h1"
  >
    153
  </h1>
  <h3
    class="MuiTypography-root MuiTypography-h3"
  />
</DocumentFragment>
`;
